### PR TITLE
Smoke Testing

### PR DESCRIPTION
**Pull request by superaxander**
I'm making this a PR so that I can leave a review, not for it to be merged yet.

### CODE REVIEW

File: examples/concepts/basic/Unsat.java [hunk 1]
```diff
@@ -8,12 +8,9 @@
...
 public class Unsat {
 
-  /*[/expect unsatisfiable]*/
   /*@
     requires 1==0;
   @*/
-  public void bad(){
-  }
-  /*[/end]*/
+  public void bad();
```
Comment (superaxander) [inline examples/concepts/basic/Unsat.java:14]: Why does this not work anymore?

File: examples/concepts/perf/postcond/PERF-POST-old-forall.pvl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+//:: cases PERF_POST_old_forall
+//:: tools silicon
+//:: verdict Pass
+
+// Performance: postcondition with \old inside a forall range.
+// \old is left in place by eliminateOldInTriggers (only strips from trigger positions).
+// Combines quantifier reasoning with pre-state reference — the pattern from ArrayList.java
+// that previously caused Viper trigger consistency errors.
+requires n >= 0;
+ensures (\forall int i = 0 .. \old(n); i >= 0 && i < n + 1);
```
Comment (superaxander) [inline examples/concepts/perf/postcond/PERF-POST-old-forall.pvl:10]: Since arguments cannot be changed by methods `\old(n) == n` in the postcondition. `\old` only does something if the value is on the heap, i.e. if you also need permission for it.

File: examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-22-old-in-invariant-body-pass.pvl [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+//:: tools silicon
+//:: verdict Pass
+
+// TC-LI-22: Loop invariant contains \old(e) in the body (not in a trigger).
+// \old(n) refers to n at method entry. The invariant i >= 0 && n == \old(n)
+// is satisfiable (e.g. i=0, n=5). Sat check must NOT fire.
+requires n >= 0;
+void f(int n) {
+    int i = 0;
+    loop_invariant i >= 0 && n == \old(n);
```
Comment (superaxander) [inline examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-22-old-in-invariant-body-pass.pvl:11]: This has the same issue, `\old(n)` is always equivalent to `n` (even if n has changed) because `\old` only applies to heap values.

File: examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-23-old-in-invariant-range-bound-pass.pvl [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+//:: verdict Pass
+
+// TC-LI-23: \old appears in the upper bound of a \forall range inside the
+// invariant — the case that broke the old eliminateOld approach because
+// variable renaming in the range bound clashed with trigger generation.
+// The invariant is satisfiable, so the sat check must NOT fire.
+requires n >= 0;
+void f(int n) {
+    int i = 0;
+    loop_invariant (\forall int k = 0 .. \old(n); k >= 0);
```
Comment (superaxander) [inline examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-23-old-in-invariant-range-bound-pass.pvl:12]: Idem

File: examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-24-old-in-invariant-unsat-fail.pvl [hunk 5]
```diff
@@ -0,0 +1,12 @@
+//:: cases TC_LI_24_OldInInvariantUnsatFail
+//:: tools silicon
+//:: verdict Fail
+
+// TC-LI-24: Loop invariant containing \old is unsatisfiable.
+// \old(n) > n && n > \old(n) can never hold simultaneously.
+// The sat check must fire with invariantUnsatisfiable.
+requires n >= 0;
+void f(int n) {
+    loop_invariant \old(n) > n && n > \old(n);
```
Comment (superaxander) [inline examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-24-old-in-invariant-unsat-fail.pvl:10]: Idem

File: examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-3-dead-after-unlock-gap.pvl [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+lock_invariant Perm(val, write) ** val > 5 ** val < 3;
+class UnlockDeadCaughtClass {
+    int val;
+
+    requires held(this) ** Perm(val, write) ** val > 5 ** val < 3;
+    void release_only() {
+        // State: false (val > 5 ∧ val < 3 is impossible).
+        // No 'lock this' here → no Refute from the lock expansion.
+        unlock this;
```
Comment (superaxander) [inline examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-3-dead-after-unlock-gap.pvl:13]: I'm not sure if doing the check on `unlock` makes sense since it only ever removes assumptions. Here we're just triggering the precondition not satisfiable check right?

File: examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-5-lockdead-without-invunsat.pvl [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+//   TC-LK-2  →  lockInvariantUnsatisfiable + lockDead  (invariant itself is bad)
+//   TC-LK-5  →  lockDead only               (invariant OK; context made state dead)
+lock_invariant Perm(val, write) ** val > 0;
+class LiveInvDeadContext {
+    int val;
+    int flag;
+
+    requires committed(this) ** Perm(flag, write) ** flag == 1;
+    void method() {
+        assume flag == 2;
```
Comment (superaxander) [inline examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-5-lockdead-without-invunsat.pvl:28]: I think in these lock examples you haven't got a case (unless I missed it) where locking causes the unsatisfiability. You could change this example to instead have `1\2` permission to flag in the lock and in the precondition, then the lock could have `flag == 2` in the invariant.

File: examples/concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-10-requires-narrows-ensures-sat.pvl [hunk 8]
```diff
@@ -0,0 +1,10 @@
+//:: cases TC_PST_10_RequiresNarrowsEnsuresSat
+//:: tools silicon
+//:: verdict Pass
+
+// TC-PST-10: Non-trivial precondition requires x > 10 combined with ensures x > 0.
+// The postcondition x > 0 is satisfiable on its own (e.g., x=1) — checker must not fire.
+// Normal verification also passes: x > 10 implies x > 0, body leaves x unchanged.
+requires x > 10;
+ensures \result  0;
```
Comment (superaxander) [inline examples/concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-10-requires-narrows-ensures-sat.pvl:9]: Missing operator

File: examples/verifythis/2024/challenge2.pvl [hunk 9]
```diff
@@ -253,6 +253,7 @@ class Main {
     requires Perm(executed[task], 1\2) ** !executed[task];
     void execute(int i, int task) {
         lock this;
+        //----------------------------no in fragma-c-------------
```
Comment (superaxander) [inline examples/verifythis/2024/challenge2.pvl:256]: ?

File: src/main/vct/main/stages/Transformation.scala [hunk 10]
```diff
@@ -435,12 +445,15 @@ case class SilverTransformation(
         GivenYieldsToArgs,
         CheckProcessAlgebra,
         EncodeCurrentThread,
+        CheckLockInvariantSatisfiability,
```
Comment (superaxander) [inline src/main/vct/main/stages/Transformation.scala:448]: This should have a comment stating it has to be before EncodeIntrinsicLock. Also maybe you want to make a flag to turn this one on and off too?

File: src/main/vct/main/stages/Transformation.scala [hunk 11]
```diff
@@ -485,10 +498,12 @@ case class SilverTransformation(
         ResolveExpressionSideEffects,
         EncodeTryThrowSignals,
         MonomorphizeClass,
+        // DetectDeadCode is placed before RefuteToInvertedAssert above
```
Comment (superaxander) [inline src/main/vct/main/stages/Transformation.scala:501]: I don't think this comment is necessary here

File: src/main/vct/main/stages/Transformation.scala [hunk 12]
```diff
@@ -485,10 +498,12 @@ case class SilverTransformation(
...
         // No more classes
         ClassToRef,
         HeapVariableToRef,
         CheckContractSatisfiability.withArg(checkSat),
+        CheckPostconditionSatisfiability.withArg(checkPostSat),
```
Comment (superaxander) [inline src/main/vct/main/stages/Transformation.scala:506]: Perhaps the functionality of CheckContractSatisfiability and CheckPostconditionSatisfiability could be merged (the name of the former implies it checks the whole contract)

File: src/rewrite/vct/rewrite/util/WellDefinednessConditions.scala [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+/** Collect the well-definedness conditions needed to evaluate an expression in a
+  * specification. For each pure function call in the expression, the function's
+  * precondition (with actual arguments substituted for formals) is included.
+  * Deeper conditions come before shallower ones so that Viper can evaluate each
+  * condition in the context established by prior ones.
+  */
+object WellDefinednessConditions {
+
+  def collect[G](expr: Expr[G], visited: Set[scala.Any] = Set.empty): Seq[Expr[G]] = {
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/util/WellDefinednessConditions.scala:16]: You could use AbstractFunction here instead of Any

File: src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+  override def desc: String =
+    "Check that lock invariants are not internally contradictory (i.e. unsatisfiable)."
+
+  case class LockInvariantUnsatisfiableBlame(stmt: Statement[_])
+      extends Blame[RefuteFailed] {
+    override def blame(error: RefuteFailed): Unit =
+      stmt match {
+        case lock: Lock[_]     => lock.blame.blame(LockInvariantUnsatisfiable(lock))
+        case unlock: Unlock[_] => unlock.blame.blame(LockInvariantUnsatisfiable(unlock))
+        case _                 =>
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala:24]: Put a ??? after this such that we crash if there is another type of statement here (since that should never happen)

File: src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+        Block[Post](Seq[Statement[Post]](
+          Inhale(dispatch(inv)),
+          Refute(ff)(LockInvariantUnsatisfiableBlame(stmt)),
+        )),
+        tt,
+      )(PanicBlame("FramedProof with trivial pre/post cannot fail")),
+      None,
+    )(PanicBlame("Extract without termination measure cannot fail"))
+
+  override def dispatch(stat: Statement[Pre]): Statement[Post] =
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala:56]: Instead of dispatching Statements to find lock/unlock statements, you could also dispatch on Declarations to find ByReferenceClasses. That would mean you don't have to keep track of the checkedClasses thing.

File: src/rewrite/vct/rewrite/CheckLoopInvariantSatisfiability.scala [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    if (!doCheck) stat.rewriteDefault()
+    else stat match {
+      case loop @ Loop(_, _, _, li @ LoopInvariant(inv, _), _) =>
+        implicit val o: Origin = li.o.where(prefix = "checkInvSat")
+        Block[Post](Seq[Statement[Post]](
+          Extract(
+            FramedProof(
+              tt,
+              {
+                val wdConditions = WellDefinednessConditions.collect(inv)
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/CheckLoopInvariantSatisfiability.scala:39]: If I looked at this WellDefinednessConditions class correctly I think it is collecting all the precondition of the functions that are called, this will also contain permissions which are not safe to simply copy in here.

You should filter the conditions such that only ones with "pure" expressions are kept (i.e. that do not refer to the heap). Additionally you don't have to look at the preconditions of all the functions that are called, taking the preconditions of the current function is sufficient.

If there are welldefinedness errors left in this loop invariant they should be ignored since the user will otherwise be warned about it twice (since it will also happen when the invariant is evaluated normally).

Comment (superaxander) [reply src/rewrite/vct/rewrite/CheckLoopInvariantSatisfiability.scala:39]: This is the same as the postcondition check where you have already implemented this filtering.

File: src/rewrite/vct/rewrite/CheckPostconditionSatisfiability.scala [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+      expr: Expr[Pre],
+      applicable: ContractApplicable[Pre],
+      resultVar: Variable[Pre],
+  )(implicit o: Origin): Expr[Pre] =
+    Substitute[Pre](Map(
+      Result[Pre](applicable.ref)(o) -> Local[Pre](resultVar.ref)(o)
+    )).dispatch(expr)
+
+  private def isNonHeap(expr: Expr[Pre]): Boolean =
+    expr match {
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/CheckPostconditionSatisfiability.scala:92]: You might want to allow `Select(c, l, r)` and `Implies(c, r)` expressions where `isNonHeap(c)` such that if someone does `a > 0 ==> (Perm(blah.x, write) ** b > 0)` that you can still get `a > 0 ==> b > 0` as a condition.
To do this you would also need to change this function a bit such that you return the non-heap part of the input expression.

I guess by not doing this all that happens is that the smoke test doesn't work for such cases (since the wellformedness error will stop the reporting of the assert false).

File: src/rewrite/vct/rewrite/DetectDeadCode.scala [hunk 18]
```diff
@@ -0,0 +1,222 @@
...
+    if (methodBlame.topOption.isEmpty) node.rewriteDefault()
+    else node match {
+      case block @ ParBlock(_, _, _, _, _, content) =>
+        implicit val o: Origin = block.o
+        // par block creates a fresh scope with the range constraint
+        block.rewrite(content =
+          instrumentBody(block, "parallel block body", content))
+      case other => other.rewriteDefault()
+    }
+}
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/DetectDeadCode.scala:222]: The missing newline at the end of the file here reminds me that you should run the `util/gitconfig.sh` script to set your Git to run the automatic formatter whenever you do a commit.

File: src/rewrite/vct/rewrite/EncodeProofHelpers.scala [hunk 19]
```diff
@@ -12,10 +12,11 @@ case object EncodeProofHelpers extends RewriterBuilderArg[Boolean] {
...
 
-  private def Indet: Origin =
-    Origin(
-      Seq(PreferredName(Seq("indet")), LabelContext("indeterminate branch"))
-    )
+  private var indetCounter: Int = 0
+  private def Indet: Origin = {
+    indetCounter += 1
+    Origin(Seq(PreferredName(Seq(s"indet$indetCounter")), LabelContext("indeterminate branch")))
+  }
```
Comment (superaxander) [inline src/rewrite/vct/rewrite/EncodeProofHelpers.scala:19]: What did you need this change for?

File: src/viper/viper/api/backend/silicon/Silicon.scala [hunk 20]
```diff
@@ -138,7 +138,7 @@ case class Silicon(
 
     val plugins =
       SilverPluginManager(Some(
-        Seq("viper.silver.plugin.standard.termination.TerminationPlugin")
+        Seq("viper.silver.plugin.standard.termination.TerminationPlugin", "viper.silver.plugin.standard.refute.RefutePlugin")
```
Comment (superaxander) [inline src/viper/viper/api/backend/silicon/Silicon.scala:141]: I forgot, but we probably also want to make this change in the Carbon.scala file. Carbon is the other verification backend for Viper. It would be good to test if the smoke tests work correctly for Carbon too (it may evaluate things in a different order sometimes). You can run VerCors with carbon with the flag `--backend=carbon`.

File: test/main/vct/test/integration/examples/DeadCodeOverheadSpec.scala [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+// the flags, so the output looks like:
+//   - Examples .../challenge1.pvl with flags --dev-no-dead ...  (baseline)
+//   - Examples .../challenge1.pvl with flags --dev-no-loop-inv-sat ...  (+dead)
+//   - Examples .../challenge1.pvl  (+all)
+//
+// Run with -oD to get per-test durations:
+//   java -Xss128m -Xmx4G -cp out.jar:res/universal/deps:res/universal/res \
+//     org.scalatest.tools.Runner -oD \
+//     -s vct.test.integration.examples.DeadCodeOverheadSpec
+//
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/DeadCodeOverheadSpec.scala:17]: Shorthand for this in a development environment is `bin/testSuite -s vct.test.integration.examples.DeadCodeOverheadSpec`

File: test/main/vct/test/integration/examples/DeadCodeSpec.scala [hunk 22]
```diff
@@ -0,0 +1,195 @@
...
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-6-ensures-stricter-satisfiable.pvl"
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-7-ensures-satisfiable-void.pvl"
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-8-ensures-result-range-sat.pvl"
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-10-requires-narrows-ensures-sat.pvl"
+
+  // Fail (PVL): postcondition contradicts itself — postUnsatisfiable fires (plus postFailed from normal verification)
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-2-ensures-unsat-contradiction.pvl"
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-4-ensures-result-contradiction.pvl"
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-5-ensures-perm-false.pvl"
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-9-ensures-n-gt-n.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/DeadCodeSpec.scala:77]: You cannot specify `fail` without `withCode`

File: test/main/vct/test/integration/examples/DeadCodeSpec.scala [hunk 23]
```diff
@@ -0,0 +1,195 @@
...
+
+  // Edge cases for postcondition calls
+  // \old in postcondition: sat checker must not fire, handled via eliminateOldInTriggers.
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-21-old-in-ensures-pass.pvl"
+  // \old in forall range: the exact pattern from ArrayList.java that previously crashed.
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-22-old-in-ensures-forall-range-pass.pvl"
+  // Nested calls: axioms fire unconditionally, postcondition is satisfiable.
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-22-nested-function-calls-sat.pvl"
+  // Nested calls where inner function negates — isPos(n) && isPos(negate(n)) is always false.
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-23-nested-function-calls-unsat.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/DeadCodeSpec.scala:110]: idem

File: test/main/vct/test/integration/examples/DeadCodeSpec.scala [hunk 24]
```diff
@@ -0,0 +1,195 @@
...
+  // Instance function call: Perm is in ensures, axiom fires, postcondition is satisfiable.
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-24-instance-function-call-sat.pvl"
+
+  // Heap field in postcondition
+  // Pass: body writes field to 0; postcondition Perm + n.val == 0 is satisfiable.
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-27-heap-field-write-pass.pvl"
+  // Fail: postcondition claims n.val > 0 AND n.val < 0 — field value contradiction.
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-28-heap-field-contradiction-fail.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/DeadCodeSpec.scala:118]: idem

File: test/main/vct/test/integration/examples/DeadCodeSpec.scala [hunk 25]
```diff
@@ -0,0 +1,195 @@
...
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-15-c-ensures-satisfiable.c"
+
+  // Fail (C): postcondition contradicts itself — checker must fire with postUnsatisfiable
+  vercors should fail withCode "postUnsatisfiable" using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-16-c-ensures-unsat.c"
+
+  // Postcondition calls a pure function.
+  // Pass: foo's precondition satisfies callee's requirement; postcondition is satisfiable.
+  vercors should verify using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-17-ensures-function-call-sat.pvl"
+  // Fail: function body is always false (no preconditions) — postUnsatisfiable fires (plus postFailed).
+  vercors should fail using silicon example "concepts/smoke_dead_post/cat2-7-postcondition/TC-PST-18-ensures-function-call-unsat.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/DeadCodeSpec.scala:96]: idem

File: test/main/vct/test/integration/examples/PerformanceSpec.scala [hunk 26]
```diff
@@ -0,0 +1,108 @@
...
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-loop-inv-sat") example "concepts/perf/postcond/PERF-POST-old-forall.pvl"
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-loop-inv-sat") example "concepts/perf/postcond/PERF-POST-perm.pvl"
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-loop-inv-sat") example "concepts/perf/postcond/PERF-POST-10methods.pvl"
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-loop-inv-sat") example "concepts/perf/postcond/PERF-POST-30methods.pvl"
+
+  // ── Sat vs unsat pairs ────────────────────────────────────────────────────────
+  // Each pair: same formula, one satisfiable (pass) and one contradictory (fail).
+
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-post-sat") example "concepts/perf/sat-unsat/PERF-SAT-inv-arith.pvl"
+  vercors should fail using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-post-sat") example "concepts/perf/sat-unsat/PERF-UNSAT-inv-arith.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/PerformanceSpec.scala:93]: idem

File: test/main/vct/test/integration/examples/PerformanceSpec.scala [hunk 27]
```diff
@@ -0,0 +1,108 @@
...
+
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-post-sat") example "concepts/perf/sat-unsat/PERF-SAT-inv-perm.pvl"
+  vercors should fail using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-post-sat") example "concepts/perf/sat-unsat/PERF-UNSAT-inv-perm.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/PerformanceSpec.scala:96]: idem

File: test/main/vct/test/integration/examples/PerformanceSpec.scala [hunk 28]
```diff
@@ -0,0 +1,108 @@
...
+
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-loop-inv-sat") example "concepts/perf/sat-unsat/PERF-SAT-post.pvl"
+  vercors should fail using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-loop-inv-sat") example "concepts/perf/sat-unsat/PERF-UNSAT-post.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/PerformanceSpec.scala:99]: idem

File: test/main/vct/test/integration/examples/PerformanceSpec.scala [hunk 29]
```diff
@@ -0,0 +1,108 @@
...
+
+  vercors should verify using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-post-sat", "--dev-no-loop-inv-sat") example "concepts/perf/sat-unsat/PERF-SAT-lock.pvl"
+  vercors should fail using silicon flags("--dev-no-dead", "--dev-no-sat", "--dev-no-post-sat", "--dev-no-loop-inv-sat") example "concepts/perf/sat-unsat/PERF-UNSAT-lock.pvl"
```
Comment (superaxander) [inline test/main/vct/test/integration/examples/PerformanceSpec.scala:102]: idem

File: test/main/vct/test/integration/helper/VercorsSpec.scala [hunk 30]
```diff
@@ -83,7 +83,7 @@ abstract class VercorsSpec extends AnyFlatSpec {
       LoggerFactory.getLogger("viper").asInstanceOf[Logger].setLevel(Level.OFF)
       LoggerFactory.getLogger("vct").asInstanceOf[Logger].setLevel(Level.INFO)
 
-      failAfter(Span(300, Seconds)) {
+      failAfter(Span(900, Seconds)) {
```
Comment (superaxander) [inline test/main/vct/test/integration/helper/VercorsSpec.scala:86]: Do you actually need this?

File: Failed
Comment (superaxander) [inline Failed:1]: What is this file for?

File: Get
Comment (superaxander) [inline Get:1]: Idem

File: Minimum
Comment (superaxander) [inline Minimum:1]: Idem

File: package-lock.json
Comment (superaxander) [inline package-lock.json:1]: idem

File: perf-timing.sh
Comment (superaxander) [inline perf-timing.sh:1]: This seems like a nice script that we might want to re-use in the future. You could put it in the util directory

File: examples/concepts/basic/Unsat.java (hunk 1 above)
Comment (Mjakot2022) [reply examples/concepts/basic/Unsat.java:14]: experimental purpose, will revert it

File: src/rewrite/vct/rewrite/EncodeProofHelpers.scala (hunk 19 above)
Comment (Mjakot2022) [reply src/rewrite/vct/rewrite/EncodeProofHelpers.scala:19]: theoretically for the test output readability, in practise it did not work and now I reverted it

File: src/main/vct/main/stages/Transformation.scala (hunk 10 above)
Comment (Mjakot2022) [reply src/main/vct/main/stages/Transformation.scala:448]: made flag for a lock, same pattern as for others

File: examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-5-lockdead-without-invunsat.pvl (hunk 7 above)
Comment (Mjakot2022) [reply examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-5-lockdead-without-invunsat.pvl:28]: good point, added this case to tests

File: examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-3-dead-after-unlock-gap.pvl (hunk 6 above)
Comment (Mjakot2022) [reply examples/concepts/smoke-test-dead/cat2-5-lock-invariant/TC-LK-3-dead-after-unlock-gap.pvl:13]: I agree, mostly precondition check.

File: examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-22-old-in-invariant-body-pass.pvl (hunk 3 above)
Comment (Mjakot2022) [reply examples/concepts/smoke-test-dead/cat2-4-loop-invariant/TC-LI-22-old-in-invariant-body-pass.pvl:11]: I agree, this test was incorrect. As i understand for now we decided to skip verification of the whole invarient if it contains \old() in general

File: src/main/vct/main/stages/Transformation.scala (hunk 12 above)
Comment (Mjakot2022) [reply src/main/vct/main/stages/Transformation.scala:506]: agreed in principle, but I would scope it as follow-up work

File: src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala (hunk 14 above)
Comment (Mjakot2022) [reply src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala:24]: done

File: src/rewrite/vct/rewrite/util/WellDefinednessConditions.scala (hunk 13 above)
Comment (Mjakot2022) [reply src/rewrite/vct/rewrite/util/WellDefinednessConditions.scala:16]: done

File: src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala (hunk 15 above)
Comment (Mjakot2022) [reply src/rewrite/vct/rewrite/CheckLockInvariantSatisfiability.scala:56]: That is a good advice and it seems like a more natural fit. I changed it

File: src/rewrite/vct/rewrite/CheckLoopInvariantSatisfiability.scala (hunk 16 above)
Comment (Mjakot2022) [reply src/rewrite/vct/rewrite/CheckLoopInvariantSatisfiability.scala:39]: yes, corrected it. forgot to do it earlier

File: src/rewrite/vct/rewrite/CheckPostconditionSatisfiability.scala (hunk 17 above)
Comment (Mjakot2022) [reply src/rewrite/vct/rewrite/CheckPostconditionSatisfiability.scala:92]: fair enough, made some changes

File: test/main/vct/test/integration/helper/VercorsSpec.scala (hunk 30 above)
Comment (Mjakot2022) [reply test/main/vct/test/integration/helper/VercorsSpec.scala:86]: not anymore

File: src/viper/viper/api/backend/silicon/Silicon.scala (hunk 20 above)
Comment (Mjakot2022) [reply src/viper/viper/api/backend/silicon/Silicon.scala:141]: updated the code, will run tests if I have time
